gui/src/bitcoind.rs

```diff
@@ -455,35 +443,43 @@ impl Bitcoind {
             .map_err(|e| StartInternalBitcoindError::CommandError(e.to_string()))?;
 
         // We've started bitcoind in the background, however it may fail to start for whatever
-        // reason. And we need its JSONRPC interface to be available to continue. Thus wait for it
-        // to have created the cookie file, regularly checking it did not fail to start.
-        let cookie_path = internal_bitcoind_cookie_path(&bitcoind_datadir, network);
+        // reason. And we need its JSONRPC interface to be available to continue. Thus wait for
+        // the interface to be created successfully, regularly checking it did not fail to start.
         loop {
             match process.try_wait() {
                 Ok(None) => {}
                 Err(e) => log::error!("Error while trying to wait for bitcoind: {}", e),
                 Ok(Some(status)) => {
                     log::error!("Bitcoind exited with status '{}'", status);
-                    return Err(StartInternalBitcoindError::CookieFileNotFound(
-                        cookie_path.to_string_lossy().into_owned(),
-                    ));
+                    return Err(StartInternalBitcoindError::ProcessExited(status));
                 }
             }
-            if cookie_path.exists() {
-                log::info!("Bitcoind seems to have successfully started.");
-                break;
+            match liana::BitcoinD::new(&config, "internal_bitcoind_start".to_string()) {
+                Ok(_) => {
+                    log::info!("Bitcoind seems to have successfully started.");
+                    return Ok(Self {
+                        config,
+                        _process: Arc::new(process),
+                    });
+                }
+                Err(liana::BitcoindError::CookieFile(_)) => {
+                    // This is only raised if we're using cookie authentication.
+                    // Assume cookie file has not been created yet and try again.
+                }
```

Comment:
It would only be required for existing wallets that use managed bitcoind. Perhaps at some point, we can migrate those wallets to use `rpcauth`. Here, we just use whichever auth is in the config file.

Reply:
Ah right it's in the config file! So this won't fix the cookie issues for existing wallets..